docs/csharp/programming-guide/classes-and-structs/codesnippet/CSharp/named-and-optional-arguments_1.cs

```diff
@@ -3,22 +3,27 @@ class NamedExample
         static void Main(string[] args)
         {
             // The method can be called in the normal way, by using positional arguments.
-            Console.WriteLine(CalculateBMI(123, 64));
+            PrintOrderDetails("Foo", 31, true);
```

Comment:
One aspect of the previous sample that I liked was that the arguments had the same type. That showed that named arguments can be used to disambiguate arguments when calling a method.

A great example for this is throwing a `ArgumentException` or an `ArgumentNullException`. The `paramName` and `Message` arguments are easily reversed.

Reply:
Good point, I'll change up the example to contain this.